modules/nf-core/fastp/main.nf

```diff
@@ -30,6 +31,13 @@ process FASTP {
     def prefix = task.ext.prefix ?: "${meta.id}"
     def adapter_list = adapter_fasta ? "--adapter_fasta ${adapter_fasta}" : ""
     def fail_fastq = save_trimmed_fail && meta.single_end ? "--failed_out ${prefix}.fail.fastq.gz" : save_trimmed_fail && !meta.single_end ? "--unpaired1 ${prefix}_1.fail.fastq.gz --unpaired2 ${prefix}_2.fail.fastq.gz" : ''
+    def out_fq = ""
+    if ( save_trimmed_pass ) {
+        out_fq = meta.single_end ? "--out1 ${prefix}.fastp.fastq.gz" : "--out1 ${prefix}_1.fastp.fastq.gz --out2 ${prefix}_2.fastp.fastq.gz"
+        out_fq = task.ext.args?.contains('--interleaved_in') ? "${prefix}.fastp.fastq.gz" : out_fq
```

Comment:
```suggestion
        out_fq = task.ext.args?.contains('--interleaved_in') ? "${prefix}.fastp.fastq.gz" : ''
```

Reply:
this would not give the proper result. I basically want line 37 to overwrite 36 only when `task.ext.args?.contains('--interleaved_in')`. it should not evaluate to `''`.

Reply:
rewrote logic for clarity here